misc/protobuf-ballerina/src/main/java/org/ballerinalang/protobuf/utils/BalFileGenerationUtils.java

```diff
@@ -0,0 +1,203 @@
+        try (InputStream targetStream = new FileInputStream(initialFile)) {
+            DescriptorProtos.FileDescriptorSet set = DescriptorProtos.FileDescriptorSet.parseFrom(targetStream);
+            if (set.getFileList().size() > 0) {
+                return set.getFile(0).toByteArray();
+            }
+        } catch (IOException e) {
+            throw new BalGenToolException("Error reading generated descriptor file '" + descriptorPath + "'.", e);
+        }
+        return new byte[0];
+    }
+    
+    /**
+     * Used to clear the temporary files.
+     *
+     * @param file file to be deleted
+     */
+    public static void delete(File file) {
+        if ((file != null) && file.exists() && file.isDirectory()) {
+            String files[] = file.list();
+            if (files != null) {
+                if (files.length != 0) {
+                    for (String temp : files) {
+                        File fileDelete = new File(file, temp);
+                        if (fileDelete.isDirectory()) {
+                            delete(fileDelete);
+                        }
+                        if (fileDelete.delete()) {
+                            LOG.debug("Successfully deleted file " + file.toString());
+                        }
+                    }
+                }
+            }
+            if (file.delete()) {
+                LOG.debug("Successfully deleted file " + file.toString());
+            }
+            if ((file.getParentFile() != null) && (file.getParentFile().delete())) {
+                LOG.debug("Successfully deleted parent file " + file.toString());
+            }
+        } else if (file != null) {
+            if (file.delete()) {
+                LOG.debug("Successfully deleted parent file " + file.toString());
+            }
+        }
+    }
+    
+    /**
+     * Sae generated intermediate files.
+     *
+     * @param url  file URL
+     * @param file destination file location
+     */
+    public static void saveFile(URL url, String file) {
+        try (InputStream in = url.openStream(); FileOutputStream fos = new FileOutputStream(new File(file))) {
+            int length;
+            byte[] buffer = new byte[1024]; // buffer for portion of data from
+            while ((length = in.read(buffer)) > -1) {
+                fos.write(buffer, 0, length);
+            }
+            fos.close();
+            in.close();
+        } catch (IOException e) {
+            throw new BalGenToolException("Error saving file '" + file + "'.", e);
+        }
+        
+    }
+    
+    /**
+     * Grant permission to the protoc executor file.
+     *
+     * @param file protoc executor file.
+     */
+    public static void grantPermission(File file) {
+        boolean isExecutable = file.setExecutable(true);
+        boolean isReadable = file.setReadable(true);
+        boolean isWritable = file.setWritable(true);
+        if (isExecutable && isReadable && isWritable) {
```

Comment:
What if this one of this conditions fails? Don't we need to put a debug log in that case? Wouldn't it be useful?